x-pack/plugins/security_solution/public/common/components/header_section/index.tsx

```diff
@@ -51,6 +59,8 @@ export interface HeaderSectionProps extends HeaderProps {
   split?: boolean;
   stackHeader?: boolean;
   subtitle?: string | React.ReactNode;
+  toggleQuery?: (status: boolean) => void;
+  toggleStatus?: boolean;
```

Comment:
These names are a little bit confusing. I had to check the implementation to understand what they are. To me`toggleStatus` sounds like a function to toggle some status, but it is a boolean instead. `toggleQuery` is even more confusing. Why not call them `isCollapsed` and `setCollapsed`?

Reply:
Another thing related to that, having these two props as optional, it is allowed to pass only one, the component will accept it without complaint, but it will not work properly.

Maybe we could define just one prop as optional, and if passed it is required to put all we need inside (as Eui usually does) something like this:
```
collapsible?: {
  setCollapsed: (status: boolean) => void;
  isCollapsed: boolean;
}
```
This way we make sure that we have both of them or we have none.

Regarding naming, I suggested the word "collapse" instead of "toggle status" or "toggle query" because here we are in the HeaderSection, those props will only make it collapse/expand the content, nothing else, This component doesn't need to know if the queries are toggled or skipped or not, it is not its work, it just needs to shows/hides the content according to these props. 
Using these names we are kind of coupling this component with an external functionality, which makes it harder to understand what they are doing.

Reply:
> those props will only make it collapse/expand the content, nothing else

not true, the props will collapse the content _and_ skip the query

Reply:
Yes, but the "skip query" logic is outside this component. This is the `HeaderSection`, the new logic here is only about showing/hiding its content, it knows nothing about skipping queries, and that's cool. I think it is better to keep it decoupled and agnostic about what happens outside, for extensibility and for readability.